cinema/views.py

```diff
@@ -101,8 +105,24 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["post"],
+        detail=True,
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = MovieSerializer(movie, data=request.data)
```

Comment:
In the `upload_image` action, consider using `MovieImageSerializer` instead of `MovieSerializer` to handle image uploads more appropriately.